addons/temps/temps.lua

```diff
@@ -0,0 +1,348 @@
+        end
+    end
+end
+
+local function receive_item(id, data, modified, injected, blocked)
+    if (id == 0x20) and (state == 'purchase') then
+        local p = packets.parse('incoming', data)
+        check_inflight(p['Item'])
+    end
+end
+
+local function obtained_key_item(p, item_id)
+    if (math.floor(item_id/512) == p['Type']) then
+        local bit = item_id % 512
+        local n = bit % 8
+        local character = math.floor(bit/8) + 1
+        return ((math.floor(p['Key item available']:byte(character)/math.pow(2, n)) % 2) == 1)
+    else
+        return false
+    end
+end
+
+local function receive_key_items(id, data, modified, injected, blocked)
+    if (id == 0x55) and (state == 'purchase') then
+        local p = packets.parse('incoming', data)
+        for k, v in pairs(inflight) do
+            if obtained_key_item(p, k) then
+                check_inflight(k)
+            end
+        end
+    end
+end
+
+local function validate_item(item)
+    for k, v in pairs(items) do
+        if item:lower() == v.name:lower() then
+            return v.name
+        end
+    end
+    error("%s not found in items list.":format(item))
+end
+
+local function start(override)
+    if not (state == 'idle') then
+        return error("Addon is busy.")
+    elseif override then
+        force = validate_item(override)
+        if not force then
+            return
+        else
+            notice("Override provided for %s.":format(force))
+        end
+    else
+        force = nil
+    end
+    local npc = validate(true)
+    if npc then
+        local p = packets.new('outgoing', 0x01a, {
+            ["Target"] = npc.id,
+            ["Target Index"] = npc.index,
+        })
+        packets.inject(p)
+        state = 'init'
+    else
+        state = 'idle'
+    end
+end
+
+local function help()
+    windower.add_to_chat(100, help_text)
+end
+
+local function blacklist(cmd, name)
+    if not cmd then
+        return error("No blacklist command provided.")
+    elseif not S{'add', 'a', 'remove', 'r'}:contains(cmd) then
+        return error("Unknown blacklist command %s.":format(cmd))
+    elseif not name then
+        return error("No blacklist command parameter provided.")
+    end
+    local item = validate_item(name)
+    if not item then
+        return
+    elseif S{'add', 'a'}:contains(cmd) then
+        notice("Adding %s to your blacklist.":format(item))
+        settings.blacklist:add(item:lower())
+    else
+        notice("Removing %s from your blacklist.":format(item))
+        settings.blacklist:remove(item:lower())
+    end
+    settings:save()
+end
+
+local function turbo()
+    if settings.turbo then
+        notice("Disabling turbo.")
+        settings.turbo = false
+    else
+        notice("Enabling turbo.")
+        settings.turbo = true
+    end
+    settings:save()
+end
+    
+local function handle_command(...)
+    local cmd  = (...) and (...):lower()
+    local args = {select(2, ...)}
```

Comment:
You only use `args` to unpack it again later, so you could save yourself the effort by just doing `handlers[cmd](select(..., 2))` later (or, if you took my suggestion above, `handlers[cmd](...)`).